lib/url.js

```diff
@@ -442,98 +442,29 @@ function validateHostname(self, rest, hostname) {
 // Automatically escape all delimiters and unwise characters from RFC 2396.
 // Also escape single quotes in case of an XSS attack.
 // Return the escaped string.
+const escapedCodes = {
+  9: '%09', 10: '%0A', 13: '%0D', 32: '%20',
+  34: '%22', 39: '%27', 60: '%3C', 62: '%3E',
+  92: '%5C', 94: '%5E', 96: '%60', 123: '%7B',
+  124: '%7C', 125: '%7D'
+};
+// Using Array is faster than Object/Map
+const escapedCodesArr = new Array();
+for (var key in escapedCodes) {
+  escapedCodesArr[key] = escapedCodes[key];
+}
```

Comment:
Is a holey array as fast as a dense array? I expect it would be better to add empty entries instead of having holes?

Reply:
It seems that dense array is still faster than holey array: https://jsperf.com/packed-vs-holey-arrays

But the comparison above is between the array case and the original `switch` case.

Reply:
Would you be so kind and switch to the dense array in that case? :-)

I guess we do not need a new benchmark in that case even though it would of course still be nice to have the results with those.

Reply:
Sorry I don't fully understand. The code of chars in `rest` is not dense, so the array used to map them is not dense as well. Changing it to a dense array may cost a lot.

Reply:
What I meant is to use it like e.g. [util.js](https://github.com/nodejs/node/blob/39dc947409c7df19ac4f9502df8cb0d4a1b2829a/lib/util.js#L96-L112).